src/signalcorr.jl

```diff
@@ -61,34 +60,32 @@ where each column in the result will correspond to a column in `x`.
 
 The output is not normalized. See [`autocor!`](@ref) for a method with normalization.
 """
-function autocov!(r::RealVector, x::RealVector, lags::IntegerVector; demean::Bool=true)
+function autocov!(
+    r::AbstractVector,
+    x::AbstractVector,
+    lags::AbstractVector{<:Integer},
+    z=zeros(typeof(zero(eltype(x)) / 1), length(x));
+    demean::Bool=true
+    )
     lx = length(x)
     m = length(lags)
     length(r) == m || throw(DimensionMismatch())
     check_lags(lx, lags)
-
     T = typeof(zero(eltype(x)) / 1)
-    z::Vector{T} = demean ? x .- mean(x) : x
+    z .= x
```

Comment:
This now also allocates `z` if `demean=false`

Reply:
Old version allocates when demean=true, though. That is the default case, so presumably more common.

Reply:
Just to avoid changing code and test simultaneously.